spec-tests/src/test/java/org/candlepin/spec/content/ContentAccessSpecTest.java

```diff
@@ -817,6 +817,82 @@ public void shouldReturnANotModifiedReturnCodeWhenTheDataHasNotBeenUpdatedSinceD
+            .addContentToProduct(ownerKey, prod.getId(), content.getId(), true);
+        adminClient.owners().createPool(ownerKey, Pools.random(prod));
+
+        ConsumerDTO consumer1 = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ConsumerDTO consumer2 = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ConsumerDTO consumer3 = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ConsumerDTO consumer4 = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ConsumerDTO consumer5 = adminClient.consumers().createConsumer(Consumers.random(owner));
+
+        List<ConsumerDTO> consumers = List.of(consumer1, consumer2, consumer3, consumer4, consumer5);
+
+        List<Response> responseList = new ArrayList<>();
+        int maxNumberOfAttempts = 5;
+        for (int i = 0; i < maxNumberOfAttempts; i++) {
+            // Update the arch of all the consumers so that the content access payload key will be
+            // different than the key used to store the content access payload previously. This causes the
+            // regeneration of the content access payload which is what we need to cause the 429 response.
+            String newArch = StringUtil.random("arch-");
+            for (ConsumerDTO consumer : consumers) {
+                consumer.putFactsItem(Facts.Arch.key(), newArch);
+                adminClient.consumers().updateConsumer(consumer.getUuid(), consumer);
+            }
+
+            List<Callable<Response>> tasks = List.of(
+                    () -> ApiClients.ssl(consumer1).consumers()
+                            .exportCertificatesInZipFormat(consumer1.getUuid(), ""),
+                    () -> ApiClients.ssl(consumer2).consumers()
+                            .exportCertificatesInZipFormat(consumer2.getUuid(), ""),
+                    () -> ApiClients.ssl(consumer3).consumers()
+                            .exportCertificatesInZipFormat(consumer3.getUuid(), ""),
+                    () -> ApiClients.ssl(consumer4).consumers()
+                            .exportCertificatesInZipFormat(consumer4.getUuid(), ""),
+                    () -> ApiClients.ssl(consumer5).consumers()
+                            .exportCertificatesInZipFormat(consumer5.getUuid(), ""));
+
+            ExecutorService execService = Executors.newFixedThreadPool(consumers.size());
+            List<Future<Response>> futures = execService.invokeAll(tasks);
+            execService.shutdown();
+            execService.awaitTermination(10L, TimeUnit.SECONDS);
+
+            for (Future<Response> future : futures) {
+                try {
+                    responseList.add(future.get());
+                }
+                catch (ExecutionException e) {
+                    throw e.getCause();
+                }
+            }
+
+        }
+
+        List<Integer> responseCodes = responseList.stream().map(Response::getCode).toList();
```

Comment:
Does this test not trigger an ApiException on a 429? We should be able to handle this block with an assertThrows, no?

Reply:
I had to use the custom Request/Response framework we built since the openapi-generated client does not allow you to specify headers. And with that framework there is no exception throwing. The response http status code and body need to be read and asserted manually.